syft/pkg/cataloger/deb/parse_dpkg_status_test.go

```diff
@@ -242,3 +259,153 @@ func TestSourceVersionExtract(t *testing.T) {
 		})
 	}
 }
+
+func Test_parseDpkgStatus(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   *bufio.Reader
+		want    []pkg.Package
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name:    "no more packages",
+			input:   bufio.NewReader(strings.NewReader(`Package: apt`)),
+			wantErr: assert.NoError,
+		},
+		{
+			name: "duplicated key",
+			input: bufio.NewReader(strings.NewReader(`
+Package: apt
+Package: apt-get
+
+`)),
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				expected := errors.New("duplicate key discovered: Package")
+				return errors.Is(err, expected)
+			},
+		},
+		{
+			name: "no match for continuation",
+			input: bufio.NewReader(strings.NewReader(`  Package: apt
```

Comment:
Maybe just include the string here and then do `bufio.NewReader(strings.NewReader(` in the test below?